honeycomb/src/main/java/org/commonjava/o11yphant/honeycomb/HoneycombManager.java

```diff
@@ -217,4 +218,69 @@ public void endTrace()
         }
     }
 
+    public Span getActiveSpan()
+    {
+        if ( beeline != null )
+        {
+            return beeline.getActiveSpan();
+        }
+        return null;
+    }
+
+    public void addCumulativeField( Span span, String name, long elapse )
+    {
+        span.addField( name, elapse );
+
+        Map<String, Object> fields = span.getFields();
+
+        // add cumulative timing field
+        String cumulativeTimingName = CUMULATIVE_TIMINGS + "." + name;
+        Long cumulativeMs = (Long) fields.get( cumulativeTimingName );
```

Comment:
The reason I was talking about using something like a Timer or a Histogram on this is so we could get total time, total count, average time, P75 time, etc. out of the metrics and report those if we wanted to, basically for free. I'm interested in average time in addition to cumulative time for these really small data sets, since that should be basically the same as asking how long did we spent total and how long did we spend per call on this operation.

Reply:
The metrics numbers, like average, correspond to "many requests", while a span is for a single requst. If we get a Timer object and extract average from it, is it proper to insert this kind of statistics to each and every span's fields? I might miss something though.
I think we might end up with using both metrics and honeycomb, each for different purposes.